src/common/snippets/src/lowered/pass/assign_registers.cpp

```diff
@@ -214,20 +236,13 @@ bool AssignRegisters::run(LinearIR& linear_ir) {
                         child_it++;
                         k++;
                     }
-                    if (k == typed_ops.size())
+                    if (k == num_expressions)
                         OPENVINO_THROW("assign registers can't find target op in the body");
-                    switch (typed_ops[k].first) {
-                        case Generator::opRegType::vec2vec:
-                        case Generator::opRegType::vec2gpr:
-                            life_out_vec[n].insert(life_in_vec[k].begin(), life_in_vec[k].end());
-                            break;
-                        case Generator::opRegType::gpr2gpr:
-                        case Generator::opRegType::gpr2vec:
-                            life_out_gpr[n].insert(life_in_gpr[k].begin(), life_in_gpr[k].end());
-                            break;
-                    }
+                    life_out_vec[n].insert(life_in_vec[k].begin(), life_in_vec[k].end());
+                    life_out_gpr[n].insert(life_in_gpr[k].begin(), life_in_gpr[k].end());
                 }
             }
+            n++;
```

Comment:
🙃 
what is the point of range-based loop if we need to keep track of the index?
Do you think we can return an index-based loop here? Like the one on L215

Reply:
Unfortunately, `exprs` is a list and this structure doesn't support indexing by `[i]`. So I have to add the separate increment of index

Reply:
Fair enough